scripts/azureml-assets/azureml/assets/release/group_test.py

```diff
@@ -3,28 +3,68 @@
 """Python script to run group tests."""
 import argparse
 from subprocess import check_call
+from subprocess import run
 from pathlib import Path
 import azure.ai.ml
 from azure.ai.ml import MLClient
 from azure.identity import DefaultAzureCredential
+from collections import defaultdict
+import concurrent.futures
 import yaml
 import os
 import sys
+from azureml.assets.util import logger
 TEST_YML = "tests.yml"
 
+
+def run_pytest_job(job: Path, my_env: dict):
+    """Run single pytest job."""
+    p = run(f"pytest -q {job}", env=my_env, shell=True)
+    return_code = p.returncode
+    return return_code
+
+
+def run_pytest_jobs_old(pytest_jobs: dict, my_env: dict):
+    """Run multiple pytest jobs concurrently."""
+    logger.print("Start running pytest jobs")
+    submitted_jobs = defaultdict(list)
+    executor = concurrent.futures.ThreadPoolExecutor()
+    for job in pytest_jobs.keys():
+        arr = [job, my_env]
+        future = executor.submit(lambda p: run_pytest_job(*p), arr)
+        submitted_jobs[future] = pytest_jobs[job]
+    return submitted_jobs
+
+def run_pytest_jobs(pytest_jobs: dict, my_env: dict):
+    """Run multiple pytest jobs concurrently."""
+    logger.print("Start running pytest jobs")
+    submitted_jobs = defaultdict(list)
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+        futures_to_job = {executor.submit(run_pytest_job, job, my_env): job for job in pytest_jobs.keys()}
+        for future in concurrent.futures.as_completed(futures_to_job):
+            job = futures_to_job[future]
+            submitted_jobs[future] = pytest_jobs[job]
+    return submitted_jobs
```

Comment:
"submitted" doesn't make as much sense now that the jobs are completed before they're added to the dict. Also, `defaultdict` isn't needed if you don't need values to be automatically created in the dict.
```suggestion
    jobs = {}
    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
        futures_to_job = {executor.submit(run_pytest_job, job, my_env): job for job in pytest_jobs.keys()}
        for future in concurrent.futures.as_completed(futures_to_job):
            job = futures_to_job[future]
            jobs[future] = pytest_jobs[job]
    return jobs
```